module-bsp/board/rt1051/bsp/magnetometer/magnetometer.cpp

```diff
@@ -14,102 +14,106 @@
-
-            bsp::KeyCodes current_parsed       = bsp::KeyCodes::Undefined;
-            bsp::KeyCodes last_slider_position = bsp::KeyCodes::Undefined;
-            Measurements last{};
-
-            enum class LPDCM_INACTIVE_TIME
-            {
-                inactive_500us,
-                inactive_1ms,
-                inactive_5ms,
-                inactive_10ms,
-                inactive_50ms,
-                inactive_100ms,
-                inactive_500ms,
-                inactive_1s
-            };
-
-            enum class BANDWIDTH_SELECT
-            {
-                bandwidth_3500Hz = 0,
-                bandwidth_7kHz   = 1,
-                bandwidth_14kHz  = 2,
-                bandwidth_10kHz  = 4,
-                bandwidth_20kHz  = 5,
-                bandwidth_40kHz  = 6
-            };
-
-            bool isTimeToCompleteWriteDefinedForRegistry(std::uint8_t address)
-            {
-                const auto it = std::find(als31300::EEPROM_REGS.begin(), als31300::EEPROM_REGS.end(), address);
-                return it != als31300::EEPROM_REGS.end();
-            }
+            inactive_500us,
+            inactive_1ms,
+            inactive_5ms,
+            inactive_10ms,
+            inactive_50ms,
+            inactive_100ms,
+            inactive_500ms,
+            inactive_1s
+        };
+
+        enum class BANDWIDTH_SELECT
+        {
+            bandwidth_3500Hz = 0,
+            bandwidth_7kHz   = 1,
+            bandwidth_14kHz  = 2,
+            bandwidth_10kHz  = 4,
+            bandwidth_20kHz  = 5,
+            bandwidth_40kHz  = 6
+        };
+
+        union i2c_buf_t
+        {
+            std::uint8_t buf[sizeof(als31300::whole_reg_t)];
+            als31300::whole_reg_t whole_reg;
+        };
 
-            TimerHandle_t timerHandle;
-            constexpr uint16_t magnetometerPollIntervalMs = 500;
+        constexpr std::uint16_t magnetometerPollIntervalMs = 500;
 
-            void TimerHandler(TimerHandle_t xTimer)
-            {
-                if (qHandleIrq != nullptr) {
-                    uint8_t val = 0x01;
-                    xQueueSend(qHandleIrq, &val, 0);
-                }
+        I2CAddress i2cAddress = {.deviceAddress = als31300::I2C_ADDRESS, .subAddress = 0, .subAddressSize = 1};
+        std::shared_ptr<drivers::DriverI2C> i2c;
+        i2c_buf_t i2cBuffer;
+
+        xQueueHandle qHandleIrq = nullptr;
+        TimerHandle_t pollingTimer;
+
+        bsp::KeyCodes currentParsedKey   = bsp::KeyCodes::Undefined;
+        bsp::KeyCodes lastSliderPosition = bsp::KeyCodes::Undefined;
+        Measurements lastMeasurement{};
+
+        bool isTimeToCompleteWriteDefinedForRegistry(std::uint8_t address)
+        {
+            const auto it = std::find(als31300::EEPROM_REGS.begin(), als31300::EEPROM_REGS.end(), address);
+            return it != als31300::EEPROM_REGS.end();
+        }
+
+        void timerHandler([[maybe_unused]] TimerHandle_t xTimer)
```

Comment:
De facto never used in this function, so leaving the parameter just unnamed and without the attribute would be even better

Reply:
I thought about it, but in such way I get warning from the clang-tidy that all of the params should be named in function, so I chose to use the attribute.